bot/cogs/help_channels.py

```diff
@@ -620,67 +630,46 @@ async def on_message(self, message: discord.Message) -> None:
+        guild = self.bot.get_guild(constants.Guild.id)
 
-        for member, overwrite in self.available_category.overwrites.items():
-            if isinstance(member, discord.Member) and overwrite.send_messages is False:
-                log.trace(f"Resetting send permissions for {member} ({member.id}).")
+        # TODO: replace with a persistent cache cause checking every member is quite slow
+        for member in guild.members:
+            if self.is_claimant(member):
+                await self.remove_cooldown_role(member)
 
-                # We don't use the permissions helper function here as we may have to reset multiple overwrites
-                # and we don't want to enforce the permissions synchronization in each iteration.
-                await self.available_category.set_permissions(member, overwrite=None)
+    @classmethod
+    async def add_cooldown_role(cls, member: discord.Member) -> None:
+        """Add the help cooldown role to `member`."""
+        log.trace(f"Adding cooldown role for {member} ({member.id}).")
+        await cls._change_cooldown_role(member, member.add_roles(COOLDOWN_ROLE))
 
-        log.trace(f"Ensuring channels in `Help: Available` are synchronized after permissions reset.")
-        await self.ensure_permissions_synchronization(self.available_category)
+    @classmethod
+    async def remove_cooldown_role(cls, member: discord.Member) -> None:
+        """Remove the help cooldown role from `member`."""
+        log.trace(f"Removing cooldown role for {member} ({member.id}).")
+        await cls._change_cooldown_role(member, member.remove_roles(COOLDOWN_ROLE))
 
-    async def reset_claimant_send_permission(self, channel: discord.TextChannel) -> None:
-        """Reset send permissions in the Available category for the help `channel` claimant."""
-        log.trace(f"Attempting to find claimant for #{channel.name} ({channel.id}).")
-        try:
-            member = self.help_channel_claimants[channel]
-        except KeyError:
-            log.trace(f"Channel #{channel.name} ({channel.id}) not in claimant cache, permissions unchanged.")
-            return
+    @staticmethod
+    async def _change_cooldown_role(member: discord.Member, coro: t.Awaitable) -> None:
+        """
+        Change `member`'s cooldown role via awaiting `coro` and handle errors.
 
-        log.trace(f"Resetting send permissions for {member} ({member.id}).")
-        await self.update_category_permissions(self.available_category, member, overwrite=None)
-        # Ignore missing task when claim cooldown has passed but the channel still isn't dormant.
-        self.cancel_task(member.id, ignore_missing=True)
+        `coro` is intended to be `discord.Member.add_roles` or `discord.Member.remove_roles`.
+        """
+        try:
+            await coro
+        except discord.NotFound:
+            log.debug(f"Failed to change role for {member} ({member.id}): member not found")
```

Comment:
There's a slight drawback here because of the use of the `discord.Object` to represent the role: `discord.NotFound` will both be raised for the member not being found as well as the role not being found. It's (hopefully) not going to be a huge issue for us in production, but if someone forgot to set an ID for the role in a dev setting, the logger will give the misleading message that the member was not found, while it was actually the role:

```
2020-04-20 11:49:13 | bot.cogs.help_channels | DEBUG | Failed to change role for Ves Zappa#3787 (336843820513755157): member not found
```

That may be a bit misleading. Another option would be to verify if the role exists for the guild the help system is being used in during the `init_cog` process and dispatch a warning when it does not (or even disable the cog, although the role is not as essential as the categories being present).

Reply:
I've addressed this by making the function always get the role.